### ISSUE TITLE
Utility::FileWatcher watching wrong filepath when using with Utility::Tweakable on Linux

### ISSUE THREAD

**Issue by Antoine-dh** [mosra/corrade#61]
When trying to use CORRADE_TWEAKABLE() with Utility::Tweakable I get a runtime error message from Utility::FileWatcher on Linux

> Utility::FileWatcher: can't stat home/ad/CLionProjects/MagnumProject/src/MyApplication.cpp: No such file or directory, aborting watch
> Utility::Tweakable: watching for changes in home/ad/CLionProjects/MagnumProject/src/MyApplication.cpp
> 

I believe the root / was not inserted at the begin of the filepath string causing the filepath to be relative instead of absolute and the file not to be found.

**Comment by mosra** [comment]
Hi! The path is gathered from the `__FILE__` preprocessor macro, so if your buildsystem underneath does something like

```sh
cd /
g++ home/ad/CLionProjects/MagnumProject/src/MyApplication.cpp -o MyApplication
```

then you'll get the path without the initial slash in `__FILE__`. I suspect this is something CLion does. As another example, in my case, when using ninja, the paths are relative to the build directory, so instead I'd get `../src/MyApplication.cpp` in `__FILE__` (which is nice and short). OTOH with plain make, as far as I remember, the paths were always full and absolute.

There's the [Tweakable::enable(const std::string&, const std::string&)](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1Tweakable.html#ae5d1ab55b5ecb61fbb534e25ef780e7f) overload that allows you to set the path prefix explicitly, it's there to cover exactly these cases. From the top of my head, I think calling

```cpp
tweakable.enable("", "/");
```

could do the trick. Hope this helps! :)

**Comment by Antoine-dh** [comment]
Hello, thanks for you answer.
`tweakable.enable("", "/");` actually did the trick
but I still think there is an issue on your side as printing `__FILE__` gives me the correct path with the root slash

**Comment by mosra** [comment]
Oh, okay, thanks for the heads-up! :+1: Looking at the code, I think I know where the problem is (and I'm sorry about the conspiration theories above). This is what I get for testing just with Ninja, which has always relative paths :sweat_smile: 

Is it possible for you to try the following patch (and running with just the argument-less `enable()`) and tell me if it works correctly in your case?

```patch
diff --git a/src/Corrade/Utility/Tweakable.cpp b/src/Corrade/Utility/Tweakable.cpp
index 8e0f8b7c..f51d847d 100644
--- a/src/Corrade/Utility/Tweakable.cpp
+++ b/src/Corrade/Utility/Tweakable.cpp
@@ -110,7 +110,7 @@ std::pair<bool, void*> Tweakable::registerVariable(const char* const file, const
            works correctly. */
         /** @todo maybe some Directory utility for this? */
         std::string stripped = String::stripPrefix(Directory::fromNativeSeparators(file), _data->prefix);
-        if(!stripped.empty() && stripped.front() == '/')
+        if(!_data->prefix.empty() && !stripped.empty() && stripped.front() == '/')
             stripped.erase(0, 1);
 
         const std::string watchPath = Directory::join(_data->replace, stripped);

```

Thanks a lot!

**Comment by Antoine-dh** [comment]
I applied your patch and it did fix the initial issue, thank you

**Comment by mosra** [comment]
Fixed on master: edeeedbe6575822bfa6ca4ab807528d876218654. Thanks for the report and testing! :+1:
